### PR TITLE
fix(dop): Deal with situations that tasks have been completed before the iteration begins

### PR DESCRIPTION
**Pull request by littlejiancc**
#### What this PR does / why we need it:
Deal with situations that have been completed before the iteration begins

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb25JRHMiOlsxMTQ2XSwiYXNzaWduZWUiOlsiMTAwMTI2MSJdfQ%3D%3D&id=291454&iterationID=1146&pId=0&type=TASK)

#### Specified Reviewers:

/assign @Effet 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |   Deal with situations that have been completed before the iteration begins           |
| 🇨🇳 中文    |      处理在迭代开始之前就已经完成的情况        |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Effet — CHANGES_REQUESTED** [review]

File: modules/dop/component-protocol/components/requirement-task-overview/chartBlock/burnoutChart/render.go [hunk 1]
```diff
@@ -87,7 +87,30 @@ func (f *BurnoutChart) Render(ctx context.Context, c *cptype.Component, scenario
 		f.Issues = append(f.Issues, issue)
 	}
 
-	finishSumAll := 0
+	// Deal with situations that have been completed before the iteration begins
+	finishSumBeforeIterBegin := 0
+	for k, issues := range issueFinishMap {
+		if k.Before(dates[0]) {
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply modules/dop/component-protocol/components/requirement-task-overview/chartBlock/burnoutChart/render.go:93]: done

File: modules/dop/component-protocol/components/requirement-task-overview/chartBlock/burnoutChart/render.go [hunk 2]
```diff
@@ -87,7 +87,30 @@ func (f *BurnoutChart) Render(ctx context.Context, c *cptype.Component, scenario
...
+			finishSumBeforeIterBegin += func() int {
+				if h.GetBurnoutChartDimension() == "total" {
+					return len(issues)
+				}
+				sumHour := 0
```
[the remark this answers is not in the document]
Comment (littlejiancc) [reply modules/dop/component-protocol/components/requirement-task-overview/chartBlock/burnoutChart/render.go:100]: done

**Review by Effet — APPROVED** [review]

**Pull request merged**